src/file/file.rpc.ts

```diff
@@ -19,167 +27,162 @@ const packageDefinition = protoLoader.loadSync(
+
 export const serviceNames: string[] = ['', 'file.fileService'];
 export const healthCheckStatusMap = {
   '': HealthCheckResponse.ServingStatus.UNKNOWN,
   serviceName: HealthCheckResponse.ServingStatus.UNKNOWN
 };
 
-// Has the full package hierarchy
-const protoDescriptor = grpc.loadPackageDefinition(packageDefinition);
-const file_proto = protoDescriptor.file;
-
 /**
  * The FileServer class, containing all of the FileServer methods.
  */
 export class FileServer {
-  public server: any;
+
+  public server: grpc.Server;
   public grpcHealthCheck: GrpcHealthCheck;
+
   public constructor(port: string) {
     this.server = new grpc.Server();
+    this.addServices();
+    this.server.bind(`0.0.0.0:${port}`, grpc.ServerCredentials.createInsecure());
+  }
 
+  private addServices () {
     // Register the health service
     this.grpcHealthCheck = new GrpcHealthCheck(healthCheckStatusMap);
     this.server.addService(HealthService, this.grpcHealthCheck);
 
-    this.server.addService(file_proto.FileService.service, {
-      GenerateKey: this.generateKey,
-      CreateUpload: this.createUpload,
-      UpdateUploadID: this.updateUpload,
-      GetUploadByID: this.getUploadByID,
-      DeleteUploadByID: this.deleteUploadByID,
-      GetFileByID: this.getFileByID,
-      GetFileByKey: this.getFileByKey,
-      GetFilesByFolder: this.getFilesByFolder,
-      CreateFile: this.createFile,
-      DeleteFile: this.deleteFile,
-      IsAllowed: this.isAllowed,
-    });
+    const fileService = {
+      GenerateKey: this.wrapper(this.GenerateKey),
+      CreateUpload: this.wrapper(this.CreateUpload),
+      UpdateUploadID: this.wrapper(this.UpdateUpload),
+      GetUploadByID: this.wrapper(this.GetUploadByID),
+      DeleteUploadByID: this.wrapper(this.DeleteUploadByID),
+      GetFileByID: this.wrapper(this.GetFileByID),
+      GetFileByKey: this.wrapper(this.GetFileByKey),
+      GetFilesByFolder: this.wrapper(this.GetFilesByFolder),
+      CreateFile: this.wrapper(this.CreateFile),
+      DeleteFile: this.wrapper(this.DeleteFile),
+      IsAllowed: this.wrapper(this.IsAllowed),
+    };
+
+    this.server.addService(file_proto.FileService.service, fileService);
+  }
+
+  private wrapper (func: Function) :
+  (call: grpc.ServerUnaryCall<Object>, callback: grpc.requestCallback<Object>) => Promise<void> {
+    return async (call: grpc.ServerUnaryCall<Object>, callback: grpc.requestCallback<Object>) => {
+      try {
+        const traceparent = call.metadata.get('elastic-apm-traceparent');
+        const transOptions = (traceparent.length > 0) ? { childOf: traceparent[0].toString() } : {};
+        apm.startTransaction(`/file.FileService/${func.name}`, 'request', transOptions);
+        const res = await func(call, callback);
+        apm.endTransaction(statusToString(grpc.status.OK));
+        callback(null, res);
+      } catch (err) {
+        const validatedErr = validateGrpcError(err);
+        const errName = statusToString(validatedErr.code);
```

Comment:
change the usage of the function as we talked and as described in `src/utils/errors/grpc.status.ts `